shotover-proxy/tests/helpers/mod.rs

```diff
@@ -21,44 +22,49 @@ pub struct ShotoverManager {
+    #[allow(dead_code)]
     pub fn from_topology_file(topology_path: &str) -> ShotoverManager {
-        let opts = ConfigOpts {
-            topology_file: topology_path.into(),
-            config_file: "tests/helpers/config.yaml".into(),
-            ..ConfigOpts::default()
-        };
-        let spawn = Runner::new(opts)
-            .unwrap_or_else(|x| panic!("{x} occurred processing {topology_path:?}"))
-            .with_observability_interface()
-            .unwrap_or_else(|x| panic!("{x} occurred processing {topology_path:?}"))
-            .run_spawn();
-
-        // If we allow the tracing_guard to be dropped then the following tests in the same file will not get tracing so we mem::forget it.
-        // This is because tracing can only be initialized once in the same execution, secondary attempts to initalize tracing will silently fail.
-        std::mem::forget(spawn.tracing_guard);
-
-        ShotoverManager {
-            runtime: spawn.runtime,
-            runtime_handle: spawn.runtime_handle,
-            join_handle: Some(spawn.join_handle),
-            trigger_shutdown_tx: spawn.trigger_shutdown_tx,
-        }
+        ShotoverManager::from_topology_file_inner(topology_path, true)
     }
 
-    #[allow(dead_code)] // to make clippy happy
+    #[allow(dead_code)]
     pub fn from_topology_file_without_observability(topology_path: &str) -> ShotoverManager {
+        ShotoverManager::from_topology_file_inner(topology_path, false)
+    }
+
+    fn from_topology_file_inner(topology_path: &str, observability: bool) -> ShotoverManager {
+        // There is a race condition between the call to take_hook and set_hook but not much we can do.
+        let prev_hook = std::panic::take_hook();
+        let (tx, panic_occured_rx) = mpsc::sync_channel(10);
+        std::panic::set_hook(Box::new(move |panic_info| {
+            // This panic hook will outlive the Receiver.
+            // so we should just ignore failures to send as that is expected once the Receiver drops.
+            let _ = tx.try_send(());
+
+            prev_hook(panic_info)
+        }));
+
         let opts = ConfigOpts {
             topology_file: topology_path.into(),
             config_file: "tests/helpers/config.yaml".into(),
             ..ConfigOpts::default()
         };
-        let spawn = Runner::new(opts)
-            .unwrap_or_else(|x| panic!("{x} occurred processing {topology_path:?}"))
-            .run_spawn();
+        let spawn = if observability {
+            Runner::new(opts)
+                .unwrap_or_else(|x| panic!("{x} occurred processing {topology_path:?}"))
+                .with_observability_interface()
+                .unwrap_or_else(|x| panic!("{x} occurred processing {topology_path:?}"))
+                .run_spawn()
+        } else {
+            Runner::new(opts)
+                .unwrap_or_else(|x| panic!("{x} occurred processing {topology_path:?}"))
+                .run_spawn()
+        };
```

Comment:
Looked a bit odd to have the same message repeated three times, but the alternative I could come up with is a bit messy, and your existing code has the benefit of distinguishing which line the error was from. 👍 

```
        let spawn = if observability {
            Runner::new(opts)
            .and_then(|r| r.with_observability_interface())
        } else {
            Runner::new(opts)
        }.map(|r| r.run_spawn())
        .unwrap_or_else(|x| panic!("{x} occurred processing {topology_path:?}"));
```